### ISSUE TITLE
[Linter] `prefer_padding_over_margin_container` lint rule

### ISSUE THREAD

**Issue by ricardodalarme** [dart-lang/sdk#56987]
### TLDR

Add lint rule to avoid using `Container` with only a `margin` property.

### Motivation

Using `Container` with only a `margin` property introduces an unnecessary widget in the widget tree, as `Container` adds a range of unused functionalities (decoration, padding, alignment) in this case. Replacing `Container` with `Padding` improves performance and makes the widget tree more readable. `Padding` is also a constant widget, which tends to be more performative.

### Summary

Encourage developers to replace `Container` widgets that only define the `margin` property with `Padding` widgets, which are optimized and provide a similar functionality with a reduced widget footprint. The rule could be added either as a new lint or as an enhancement to the existing `avoid_unnecessary_containers` rule.
### Lint Rule Implementation

- **Rule Name:** `prefer_padding_over_margin_container`
- **Description:** Warns when a `Container` is used with only a `margin` property, suggesting the use of `Padding` instead.
- **Severity Level:** **WARNING** (since it is a performance and readability improvement, not a breaking issue)

##### Bad

```dart
Container(
  margin: EdgeInsets.all(8),
  child: Text("Hello, World!"),
);
````

##### Good

```dart
Padding(
  padding: EdgeInsets.all(8),
  child: Text("Hello, World!"),
);
```
